### PR TITLE
avoid rewrapping already wrapped compiler/linker command with RPATH wrapper script

### PR DESCRIPTION
**Pull request by boegel**
fix for #2010

cc @pescobar

### CODE REVIEW

**Comment by boegel** [comment]
@wpoely86, @rjeschmi: please review?

File: easybuild/tools/toolchain/toolchain.py [hunk 1]
```diff
@@ -727,6 +729,15 @@ def prepare_compiler_cache(self, cache_tool):
         self.cached_compilers.update(compilers)
         self.log.debug("Cached compilers (after preparing for %s): %s", cache_tool, self.cached_compilers)
 
+    @staticmethod
+    def is_rpath_wrapper(path):
+        """
+        Check whether command at specified location already is an RPATH wrapper script rather than the actual command
+        """
+        in_rpath_wrappers_dir = os.path.basename(os.path.dirname(path)) == RPATH_WRAPPERS_SUBDIR
+        calls_rpath_args = 'rpath_args.py $CMD' in read_file(path)
```
Comment (wpoely86) [inline easybuild/tools/toolchain/toolchain.py]: use hash?

Comment (boegel) [inline easybuild/tools/toolchain/toolchain.py]: The RPATH wrapper script that is in place is a templated version of `easybuild/scripts/rpath_wrapper_template.sh.in`. That doesn't mean we can't hash it, but I'd rather have a stupid yet safe check in place to really avoid wrapping an existing wrapper. We really *really* REALLY don't want to include a fork bomb in EasyBuild...

Checking a hash is really strict, you just need a single byte changed somehow to have a hash mismatch...

Comment (wpoely86) [inline easybuild/tools/toolchain/toolchain.py]: Yes, so it's very very strict, like you want?

Comment (boegel) [inline easybuild/tools/toolchain/toolchain.py]: I don't necessarily want the check to see whether the command I'm wrapping is a wrapper script already to be strict. On the contrary, I'd rather do a less strict non-hash check (like I'm doing now), which limits the chance of running into a false negative and wrapping an existing wrapper...

Comment (boegel) [inline easybuild/tools/toolchain/toolchain.py]: @wpoely86 So, do we still disagree on this?

**Comment by boegel** [comment]
Thanks for the review @wpoely86@

**Pull request merged**